app/scripts/migrations/175.ts

```diff
@@ -0,0 +1,120 @@
+import { hasProperty, isObject } from '@metamask/utils';
+import { cloneDeep } from 'lodash';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 175;
+
+const SOLANA_MAINNET_ADDRESS = 'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp';
+
+type TransactionStateEntry = {
+  transactions: unknown[];
+  next: string | null;
+  lastUpdated: number;
+};
+
+type NewTransactionsState = {
+  [accountId: string]: {
+    [chainId: string]: TransactionStateEntry;
+  };
+};
+
+/**
+ * Validate that the given entry is a valid transaction state entry.
+ *
+ * @param entry - The entry to validate.
+ * @returns True if the entry is valid, false otherwise.
+ */
+function isValidTransactionStateEntry(
+  entry: unknown,
+): entry is TransactionStateEntry {
+  return (
+    typeof entry === 'object' &&
+    entry !== null &&
+    hasProperty(entry, 'transactions') &&
+    hasProperty(entry, 'next') &&
+    hasProperty(entry, 'lastUpdated') &&
+    Array.isArray(entry.transactions) &&
+    (typeof entry.next === 'string' || entry.next === null) &&
+    typeof entry.lastUpdated === 'number'
+  );
+}
+
+/**
+ * This migration transforms the MultichainTransactionsController state structure
+ * to support per-chain transaction storage. It moves transactions from directly
+ * under the account to be nested under the chainId (Solana in this case).
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly
+ * what we persist to disk.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(
+  state: Record<string, unknown>,
+): Record<string, unknown> {
+  if (
+    !hasProperty(state, 'MultichainTransactionsController') ||
+    !isObject(state.MultichainTransactionsController)
```

Comment:
Nit: Sorry I wasn't clear before, but I also didn't mean to suggest this condition be changed 😅 

If the `MultichainTransactionsController` is missing completely, that is OK. But if it's not an object, that is worth reporting because it might suggest we had a strange bug somewhere causing this state to get corrupted.

That said, I think this is extremely unlikely and we'd probably discover this corruption in other ways if it did happen, so not a blocker.